src/shims/unix/fs.rs

```diff
@@ -1026,8 +1013,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
             this.reject_in_isolation("`opendir`", reject_with)?;
             let eacc = this.eval_libc("EACCES");
-            this.set_last_error(eacc)?;
-            return Ok(Scalar::null_ptr(this));
```

Comment:
This returned a null pointer before and now you made it return -1. That's definitely wrong, please keep the old code here.